crates/noirc_evaluator/src/ssa_refactor.rs

```diff
@@ -34,23 +34,26 @@ pub(crate) fn optimize_into_acir(
     print_ssa_passes: bool,
 ) -> GeneratedAcir {
     let abi_distinctness = program.return_distinctness;
-    let ssa = ssa_gen::generate_ssa(program).print(print_ssa_passes, "Initial SSA:");
+    let mut ssa = ssa_gen::generate_ssa(program).print(print_ssa_passes, "Initial SSA:");
     let brillig = ssa.to_brillig();
-    ssa.inline_functions()
-        .print(print_ssa_passes, "After Inlining:")
-        .unroll_loops()
-        .print(print_ssa_passes, "After Unrolling:")
-        .simplify_cfg()
-        .print(print_ssa_passes, "After Simplifying:")
-        .flatten_cfg()
-        .print(print_ssa_passes, "After Flattening:")
-        .mem2reg()
-        .print(print_ssa_passes, "After Mem2Reg:")
-        .fold_constants()
-        .print(print_ssa_passes, "After Constant Folding:")
-        .dead_instruction_elimination()
-        .print(print_ssa_passes, "After Dead Instruction Elimination:")
-        .into_acir(brillig, abi_distinctness, allow_log_ops)
+    if let RuntimeType::Acir = ssa.main().runtime() {
+        ssa = ssa
+            .inline_functions()
```

Comment:
I think this check should be in the relevant passes when a function is operated on rather than over all of the SSA.

It makes sense for passes like inlining which start in main and inline from there but for passes like fold_constants which just iterate through all function I think it'd make more sense if the check was on each individual function.